app/code/core/Mage/Contacts/controllers/IndexController.php

```diff
@@ -57,6 +62,11 @@ public function indexAction()
 
     public function postAction()
     {
+        if (!$this->_validateFormKey()) {
+            $this->_redirectReferer();
+            return;
+        }
+
         $post = $this->getRequest()->getPost();
```

Comment:
In edge cases where the session has expired, it'll be nice to notify the user "Invalid Form Key. Please resubmit your request." In this case, it's necessary to repopulate the form. As an example:

```suggestion
        $post = $this->getRequest()->getPost();
        if (!$this->_validateFormKey()) {
            Mage::getSingleton('customer/session')->addError(Mage::helper('contacts')->__('Invalid Form Key. Please resubmit your request.'));
            Mage::getSingleton('customer/session')->setContactFormData($post); // To repopulate the form in block.
            $this->_redirectReferer();
            return;
        }
```

But it's even better to validate the form key within the `try catch` block; this way, the form can be repopulated whenever there is an error.

Reply:
Good idea!